### DISCUSSION TITLE
How to scale a CQ-Object

### DISCUSSION THREAD

**Discussion by Paul8043** [CadQuery/cadquery#1310] [General]
How to scale a CQ-Object without changing the dimension of the box? I am looking for the counterpart of OpenSCAD scale([xy,z]). I have searched in the online-manual, but could not find a solution there. This is a functionality needed very often, e.g. when printing a shrunk prototype with a 3D-printer.
```
# HowTo: scale a CQ-object, looking for the counterpart of OpenSCAD scale([x,y,Z])

import cadquery as cq

result = cq.Workplane().box(100,100,100).scale(0.1)            # same factor for all axis
# or 
result = cq.Workplane().box(100,100,100).scale((0.2,0.1,0.5))  # individual factor for each axis

# show_object(result,name="result",options={"alpha":0.2,"color":(255,170,0)})
```

**Comment by Paul8043** [comment]
There exists  a scale(factor) for Shapes. But I need this for a Workplane-Object. Please help.

**Comment by jmwright** [comment]
You can call scale on the wrapped object and it will be as if you called it on the Workplane itself.
```python
import cadquery as cq

box = cq.Workplane().box(10, 10, 10)
cylinder = cq.Workplane().center(10, 10).cylinder(10, 5)

box = box.val().scale(0.1)

show_object(box)
show_object(cylinder)
```
![Screenshot from 2023-04-17 08-48-36](https://user-images.githubusercontent.com/1015439/232488835-07b41c06-75b6-4e2f-919e-08614bb147bd.png)

**Reply by Paul8043** [reply]
@jmwright 
Thanks for this tip. I will use scale() for making small 3D-prototypes with a printer. OpenSCAD-porters would appreciate the second form, where each axis can be scaled down individually. This might seem to be an unusual use-case, but this is not the complete truth. Most often the second form is used to perform mirroring e.g. scale([1,-1,1])  would mirror on y-axis like mirrorY().

**Comment by jmwright** [comment]
Potentially related issues on anisotropic scaling.

https://github.com/CadQuery/cadquery/issues/638

https://github.com/CadQuery/cadquery/issues/1026

https://github.com/CadQuery/cadquery/pull/225 (that was before the switch to OCP)
